keps/sig-scheduling/5278-nominated-node-name-for-expectation/README.md

```diff
@@ -310,7 +360,10 @@ If an external component adds `NominatedNodeName`, but the scheduler picks up a
 But, if an external component updates `NominatedNodeName` that is set by the scheduler, 
 the pod could end up having different `NominatedNodeName` and `NodeName`.
 
-Probably we should clear `NominatedNodeName` when the pod is bound. (at binding api)
+We will update the logic so that:
+- `NominatedNodeName` field is cleared during `binding` call
+- [to be confirmed during API review] we fail the validation when someone attempts to set
+`NNN` for an already bound pod
```

Comment:
Do we actually need this? From our perspective, it does not make sense to set NNN for bound pods though, I'm afraid that this validation would just break some external components unnecessarily, which have already utilized NNN and have a (weird?) behavior to set NNN for bound pods for some purpose.
I'm not sure if we should take such a weird case into account, which we're not even sure if exists, though, at the same time, I'm not sure if we should explicitly ban those API operations on purpose, from now.

Reply:
Probably, what I wanted to say is: do we have any advantage to do this by taking a risk of breaking existing components that _might_ exist in this world? If someone does this, then it might make sense to them for their specific use cases. And, it's their fault that, as a result, NNN vs NodeName could be different and might be confusing. But, they should already know that, and still do that on purpose for some reason.

Reply:
My motivation was that it's misleading when NodeName and NominatedNodeName are different, no matter when they were set. This second part was addressing the case when NNN is set later.

I'm fine with saying that it's such a niche thing that we don't do that for the reasons you mentioned.

Reply:
I went ahead and already adjusted it accordingly.